pkg/loadbalancer/address_other.go

```diff
@@ -2,10 +2,20 @@
 
 package loadbalancer
 
+import "os/exec"
+
 func AddIPToInterface(ifaceName string, ip string) error {
+	err := exec.Command("ip", "addr", "add", ip, "dev", "lo").Run()
```

Comment:
```suggestion
	err := exec.Command("ip", "addr", "add", ip, "dev", ifaceName).Run()
```

Reply:
unfortunately it doesn't work:
```
failed to ensure load balancer: exit status 1
```
because in WSL2 there is only `lo`
```
❯ ifconfig
eth0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.25.124.246  netmask 255.255.240.0  broadcast 172.25.127.255
        inet6 fe80::215:5dff:feff:b7a0  prefixlen 64  scopeid 0x20<link>
        ether 00:15:5d:ff:b7:a0  txqueuelen 1000  (Ethernet)
        RX packets 76102  bytes 87555200 (87.5 MB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 59909  bytes 7699849 (7.6 MB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 646306  bytes 1683348381 (1.6 GB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 646306  bytes 1683348381 (1.6 GB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
```
and with `ifaceName` == `lo0`
```
❯ sudo ip addr add 172.20.0.3 dev lo0
Cannot find device "lo0"
```

Reply:
then do something like this
```diff
diff --git a/pkg/loadbalancer/tunnel.go b/pkg/loadbalancer/tunnel.go
index 058e106..3fd4b62 100644
--- a/pkg/loadbalancer/tunnel.go
+++ b/pkg/loadbalancer/tunnel.go
@@ -43,6 +43,11 @@ func (t *tunnelManager) setupTunnels(containerName string) error {
        }
 
        klog.V(0).Infof("setting IPv4 address %s associated to container %s", ipv4, containerName)
+       // linux and windows uses lo0
+       ifaceName := "lo0"
+       if isWSL2() {
+               ifaceName = "lo"
+       }
        err = AddIPToInterface(ifaceName, ipv4)
        if err != nil {
                return err
```

but respect the API

Reply:
`lo0` is a name of a local interface in MacOS only
even in `address_windows.go` it isn't used (it's replaced with hardcoded `loopback`):
```
err := exec.Command("netsh", "interface", "ip", "add", "address", "loopback", ip, "255.255.255.255").Run()
```
so why it is a default value for the constant? 

I suggest to move `ifaceName` to implementation specific layer (`address_*.go` files) and remove parameter from functions `AddIPToInterface` and `RemoveIPToInterface`
they can also be renamed to something like `AddIPToLocalInterface` and `RemoveIPFromLocalInterface` accordingly

Reply:
done